examples/cpp/PineAPPL.hpp

```diff
@@ -0,0 +1,257 @@
+     * @param bins bins
+     * @param key_val additional informations
+     */
+    Grid(const Lumi& lumi, const std::vector<Order>& orders, const std::vector<double>& bins, const KeyVal& key_val) {
+        this->n_lumis = lumi.count();
+        // cast orders
+        uint32_t *raw_orders = new uint32_t[4 * orders.size()];
+        for (auto it = orders.cbegin(); it != orders.cend(); ++it, ++this->n_orders){
+            raw_orders[4*this->n_orders + 0] = it->alphas;
+            raw_orders[4*this->n_orders + 1] = it->alpha;
+            raw_orders[4*this->n_orders + 2] = it->logxir;
+            raw_orders[4*this->n_orders + 3] = it->logxif;
+        }
+        //this->n_bins = bins.size();
+        // cast bins
+        double *raw_bins = new double[bins.size()];
+        for (auto it = bins.cbegin(); it != bins.cend(); ++it, ++this->n_bins)
+            raw_bins[this->n_bins] = *it;
+        // now, we got all we need
+        this->raw = pineappl_grid_new(lumi.raw, this->n_orders, raw_orders, this->n_bins - 1, raw_bins, key_val.raw);
+        // clean up
+        delete[] raw_orders;
+        delete[] raw_bins;
+    }
+
+    /** @brief destructor */
+    ~Grid(){ pineappl_grid_delete(this->raw); }
+
+    /**
+     * @brief Fill grid for the given parameters
+     * @param x1 first momentum fraction
+     * @param x2 second momentum fraction
+     * @param q2 scale
+     * @param order order index
+     * @param observable observable value
+     * @param lumi luminosity index
+     * @param weight weight
+     */
+    void fill(double x1, double x2, double q2, size_t order, double observable, size_t lumi, double weight) const {
+        pineappl_grid_fill(this->raw, x1, x2, q2, order, observable, lumi, weight);
+    }
+
+    /**
+     * @brief perform a convolution of the grid with PDFs
+     * @param pdg_id particle ID
+     * @param pdf PDF
+     * @param xi_ren renormalization scale variation
+     * @param xi_fac factorization scale variation
+     * @return prediction for each bin
+     * @return prediction for each bin
+     */
+    std::vector<double> convolute_with_one(int32_t pdg_id, LHAPDF::PDF *pdf, double xi_ren = 1.0, double xi_fac = 1.0 ) const {
+        std::vector<bool> order_mask(this->n_orders, true);
+        std::vector<bool> lumi_mask(this->n_lumis, true);
+        return this->convolute_with_one(pdg_id, pdf, order_mask, lumi_mask, xi_ren, xi_fac);
+    }
+
+    /**
+     * @brief perform a convolution of the grid with PDFs
+     * @param pdg_id particle ID
+     * @param pdf PDF
+     * @param order_mask order mask
+     * @param lumi_mask luminosity mask
+     * @param xi_ren renormalization scale variation
+     * @param xi_fac factorization scale variation
+     * @return prediction for each bin
+     */
+    std::vector<double> convolute_with_one(int32_t pdg_id, LHAPDF::PDF *pdf, const std::vector<bool>& order_mask, const std::vector<bool>& lumi_mask, double xi_ren = 1.0, double xi_fac = 1.0) const {
```

Comment:
and give `order_mask` and `lumi_mask` default parameters (empty vectors). Check if each of them are empty and if they are, pass `nullptr` instead below to `pineappl_grid_convolute_with_one`, which is understood as 'enable everything'.